### DISCUSSION TITLE
Why isn't an account id guaranteed to be a number?

### DISCUSSION THREAD

**Discussion by YOCKOW** [mastodon/mastodon#28454] [Q&A] [answered]
I read ["API Entities > Account > Attributes > id" in the Mastodon docs](https://docs.joinmastodon.org/entities/Account/#id) and found the following part:

> id
> Description: The account id.
> Type: String (cast from an integer, but **not guaranteed to be a number**)

IIUC, [`account_id` is saved as a `bigint` in DB](https://github.com/mastodon/mastodon/blob/d29b1cca2e1dd9f38a12086ca8010fafbb0cac33/db/schema.rb).
Why isn't an account id guaranteed to be a number?

**Comment by Gargron** [comment, marked as answer]
The reason it's encoded as a string is to avoid integer overflow in JavaScript. In Mastodon it's pretty much always going to be a number. However there are third-party platforms that emulate the Mastodon API to let people use familiar Mastodon apps with them (e.g. Pleroma, Akkoma, GoToSocial) that have different internal data structures. At the end of the day, you don't need to do any number things like addition or multiplication with the id, so just treating it as an opaque string works.

**Reply by YOCKOW** [reply]
Thank you for your quick and comprehensible reply.
I absolutely missed other platforms.
